cmd/sysl/cmd_databasescript_mod.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/spf13/afero"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	sysl "github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/sirupsen/logrus"
+)
+
+func GenerateModDatabaseScripts(scriptParams *CmdDatabaseScriptMod,
+	logger *logrus.Logger) (map[string]string, error) {
+	outmap := make(map[string]string)
+
+	logger.Debugf("project: %v\n", scriptParams.project)
+	logger.Debugf("title: %s\n", scriptParams.title)
+	logger.Debugf("output: %s\n", scriptParams.output)
+	logger.Debugf("source: %s\n", scriptParams.orgSource)
+	logger.Debugf("output: %s\n", scriptParams.newSource)
+
+	modelOld, _, err1 := LoadSyslModule(scriptParams.root, scriptParams.orgSource, afero.NewOsFs(), logger)
+	if err1 != nil {
+		return nil, err1
+	}
+	modelNew, _, err2 := LoadSyslModule(scriptParams.root, scriptParams.newSource, afero.NewOsFs(), logger)
+	if err2 != nil {
+		return nil, err2
+	}
+	// The "project" app that specifies the data models to be built
+	var appOld, appNew *sysl.Application
+	var existsOld, existsNew bool
+	appOld, existsOld = modelOld.GetApps()[scriptParams.project]
+	appNew, existsNew = modelNew.GetApps()[scriptParams.project]
+	if !(existsOld || existsNew) {
+		return nil, fmt.Errorf("project not found in sysl for either new or old script")
+	}
+
+	epname := "Relational-Model"
+	endptOld := appOld.GetEndpoints()[epname]
+	endptNew := appNew.GetEndpoints()[epname]
+	outputDir := scriptParams.output
+	if strings.Contains(outputDir, "%(epname)") {
+		of := MakeFormatParser(scriptParams.output)
+		outputDir = of.FmtOutput(scriptParams.project, epname, "", nil)
+	}
+	generateModDatabaseScripts(outmap, modelOld, modelNew, endptOld.GetStmt(), endptNew.GetStmt(),
+		scriptParams.title, scriptParams.project, outputDir)
+	return outmap, nil
+}
+
+func generateModDatabaseScripts(outmap map[string]string, modOld *sysl.Module, modNew *sysl.Module,
+	stmtsOld []*sysl.Statement, stmtsNew []*sysl.Statement, title, project, outDir string) {
+	appsOld := modOld.GetApps()
+	appsNew := modNew.GetApps()
+	var stringBuilder strings.Builder
```

Comment:
should not be created here

Reply:
Code has changed and this comment has been addressed